src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,77 @@
-import React from 'react';
+import React, { FormEvent, useState } from 'react';
+import { Comment } from '../types/Comment';
+import classNames from 'classnames';
+import { Post } from '../types/Post';
+
+type Props = {
+  onAdd: (val: Omit<Comment, 'id'>) => Promise<void>;
+  selectedPost: Post | null;
+};
+
+export const NewCommentForm: React.FC<Props> = ({ onAdd, selectedPost }) => {
+  const [isSubmiting, setIsSubmiting] = useState(false);
+
+  const [name, setName] = useState('');
+  const [isErrorName, setIsErrorName] = useState(false);
+
+  const [email, setEmail] = useState('');
+  const [isErrorEmail, setIsErrorEmail] = useState(false);
+
+  const [comment, setComment] = useState('');
+  const [isErrorComment, setIsErrorComment] = useState(false);
+
+  const handleNameChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setName(e.target.value);
+    setIsErrorName(false);
+  };
+
+  const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setEmail(e.target.value);
+    setIsErrorEmail(false);
+  };
+
+  const handleCommentChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setComment(e.target.value);
+    setIsErrorComment(false);
+  };
+
+  const handleReset = () => {
+    setComment('');
+    setEmail('');
+    setName('');
+    setIsErrorComment(false);
+    setIsErrorEmail(false);
+    setIsErrorName(false);
+  };
+
+  const handleSubmit = (e: FormEvent) => {
+    e.preventDefault();
+
+    setIsErrorName(!name);
+    setIsErrorEmail(!email);
+    setIsErrorComment(!comment);
+
+    if (!name || !email || !comment) {
+      return;
+    }
+
+    setIsSubmiting(true);
+    onAdd({
+      name: name.trim(),
+      email: email.trim(),
+      body: comment.trim(),
+      postId: selectedPost?.id || 0,
+    }).finally(() => {
+      setComment('');
```

Comment:
It's not a good practice to call `setState` several times in one function call (method). Try to group these calls into a single one for clarity of the code. This is not a critical issue, but it would improve the readability of your code.